### ISSUE TITLE
rke v1.4.3-rc2 fails to provision a k8s v1.24.8-rancher1-1 and v1.23.14

### ISSUE THREAD

**Issue by vivek-shilimkar** [rancher/rke#3134]
**RKE version: v1.4.3-rc2**

**Docker version: (`docker version`,`docker info` preferred)  20.10.21**

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Type/provider of hosts: AWS**

**cluster.yml file:**
Without system_images : [cluster.txt](https://github.com/rancher/rke/files/10331950/cluster.txt)
With auto-generated `system_images` after `rke config` command: [cluster.long.txt](https://github.com/rancher/rke/files/10332010/cluster.long.txt)

**Steps to Reproduce:**

**1. With the cluster.txt file**

- Change the cluster.txt into cluster.yml.
- kubernetes_version in the cluster.yml file is set to "v1.23.14-rancher1-1" and system_images section are removed.
- Download the RKE v1.4.3-rc2, change the node details and run the rke up command using the above cluster.yml file.
- Cluster provisioning will be successful however pods will keep restarting.

**2. With the cluster.long.txt file**
- Change the cluster.long.txt into cluster.yml.
- Please note `kubernetes_version` in this file is not set. `kubernetes: rancher/hyperkube:v1.23.14-rancher1` in system_images is set while providing input to `rke config` command.
- Download the RKE `v1.4.3-rc2 `and run the `rke up` command using the above cluster.yml file.
- Alternately you can run `rke config` command, 
  - provide the node details
  - provide `v1.23.14-rancher1-1` or `v1.24.8-rancher1-1` hyperkube image (Default is `kubernetes: rancher/hyperkube:v1.25.5-rancher1`). 
  - Once the `rke config` stops asking for input, run `rke up`.
- Cluster provisioning will fail.

**Results:**

- Without system_images section in the cluster.yml cluster provisions successfully. However, Many pods keeps restarting in the cluster.

![Screenshot from 2023-01-02 19-20-41](https://user-images.githubusercontent.com/83208989/210240731-1849ceac-55c8-4c3a-ba6b-0ed39429ac3a.png)

- With system_images or cluster.yml file that gets created with `rke config` command cluster provisioning fails with following error in the log.

`FATA[0150] [controlPlane] Failed to bring up Control Plane: [Failed to verify healthcheck: Failed to check https://localhost:6443/healthz for service [kube-apiserver] on host [3.21.41.161]: Get "https://localhost:6443/healthz": Unable to access the service on localhost:6443. The service might be still starting up. Error: ssh: rejected: connect failed (Connection refused), log: E0102 07:27:29.514561       1 run.go:74] "command failed" err="failed to initialize admission: couldn't init admission plugin \"PodSecurity\": no kind \"PodSecurityConfiguration\" is registered for version \"pod-security.admission.config.k8s.io/v1\" in scheme \"vendor/k8s.io/pod-security-admission/admission/api/scheme/scheme.go:30\""] 
`

**Comment by jiaqiluo** [comment]
### case 1 With the cluster.txt file

I could not reproduce the test 1 on my end. 
Following the exact steps and using the cluster.txt from the above comment, the cluster is provisioned successfully.
```
> k version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.4", GitCommit:"872a965c6c6526caa949f0c6ac028ef7aff3fb78", GitTreeState:"clean", BuildDate:"2022-11-09T13:28:30Z", GoVersion:"go1.19.3", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.14", GitCommit:"3321ffc07d2f046afdf613796f9032f4460de093", GitTreeState:"clean", BuildDate:"2022-11-09T13:32:47Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.25) and server (1.23) exceeds the supported minor version skew of +/-1
```
```
> k get pods -A
NAMESPACE       NAME                                      READY   STATUS      RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-mvfql      0/1     Completed   0          2m59s
ingress-nginx   ingress-nginx-admission-patch-cq8cs       0/1     Completed   0          2m59s
ingress-nginx   nginx-ingress-controller-zqqzv            1/1     Running     0          2m59s
kube-system     calico-kube-controllers-fc7fcb565-lfln9   1/1     Running     0          3m35s
kube-system     canal-wbbqt                               2/2     Running     0          3m36s
kube-system     coredns-548ff45b67-rhh9s                  1/1     Running     0          3m21s
kube-system     coredns-autoscaler-d5944f655-wn48m        1/1     Running     0          3m20s
kube-system     metrics-server-5c4895ffbd-bv4fs           1/1     Running     0          3m10s
kube-system     rke-coredns-addon-deploy-job-xpsxz        0/1     Completed   0          3m29s
kube-system     rke-ingress-controller-deploy-job-dq9lw   0/1     Completed   0          3m2s
kube-system     rke-metrics-addon-deploy-job-zkqgf        0/1     Completed   0          3m13s
kube-system     rke-network-plugin-deploy-job-blkpg       0/1     Completed   0          3m40s
```

### case 2 With the cluster.long.txt file

I reproduced this error but it is actually not a bug.  It happens because  the `kubernete_version` is not set in the cluster-long.yml, therefore RKE users the default version which is 1.25 to provision the cluster, and then it runs into the API version mismatch issue.  If you run `rke -d up --config ... `, RKE logs will show that the k8s version for the cluster is 1.25. 

The cluster is up successfully after I set `kubernetes_version: "v1.23.14-rancher1-1"` 

With that being said, can QA try the test 1 on your end, and close this issue if it is not reproducible? 
cc @sowmyav27 @vivek-infracloud

**Comment by vivek-shilimkar** [comment]
@jiaqiluo Thanks for being patient with me. For some reason I encountered the error mentioned in the issue for `cluster.yaml` file.
However, today the cluster provisioning was successful.

You rightly pointed out that the missing `kubernetes_version` is the reason for unsuccessful provisioning of the cluster in `cluster.long.yaml`. After setting the `kubernetes_version` cluster provisioned successfully.

Closing the issue.
